science/hacktv/files/destdir-fix.diff

```diff
@@ -0,0 +1,33 @@
+--- Makefile.orig	2022-10-09 22:13:57.000000000 +0100
++++ Makefile	2022-10-09 23:19:38.000000000 +0100
+@@ -1,8 +1,8 @@
+ 
+-CC      := $(CROSS_HOST)gcc
+-PKGCONF := $(CROSS_HOST)pkg-config
+-CFLAGS  := -g -Wall -pthread -O3 $(EXTRA_CFLAGS)
+-LDFLAGS := -g -lm -pthread $(EXTRA_LDFLAGS)
++CC      ?= $(CROSS_HOST)gcc
++PKGCONF ?= $(CROSS_HOST)pkg-config
++CFLAGS  ?= -g -Wall -pthread -O3 $(EXTRA_CFLAGS)
++LDFLAGS ?= -g -lm -pthread $(EXTRA_LDFLAGS)
+ OBJS    := hacktv.o common.o fir.o vbidata.o teletext.o wss.o video.o mac.o dance.o eurocrypt.o videocrypt.o videocrypts.o syster.o acp.o vits.o nicam728.o test.o ffmpeg.o file.o hackrf.o
+ PKGS    := libavcodec libavformat libavdevice libswscale libswresample libavutil libhackrf $(EXTRA_PKGS)
+ 
+@@ -32,8 +32,16 @@
+ 	$(CC) $(CFLAGS) -c $< -o $@
+ 	@$(CC) $(CFLAGS) -MM $< -o $(@:.o=.d)
+ 
++# Macports wants the install phase to be "make install DESTDIR=blah" 
++# (see https://guide.macports.org/chunked/reference.phases.html ) so 
++# to allow both macports and non-macports builds, we reallly should
++# either detect macports or DESTDIR and act accordingly. However, for
++# now let's just assume it's only macports trying to build us.
++# Macports wants opt/local
+ install:
+-	cp -f hacktv $(PREFIX)/usr/local/bin/
++	mkdir -p $(DESTDIR)/bin/
++	install -c hacktv $(DESTDIR)/opt/local/bin/
```

Comment:
this path should not be hardcoded as users are free to install MacPorts in another prefix

Reply:
OK - help me here. I must have misunderstood the guidance I read doing that. I thought that DESTDIR was the very variable which (directly or indirectly) would be being controlled by a user trying to use other prefixes.  Is that not the case?  More concretely, perhaps I should ask just what should it look like? Is it like :

`install -c hacktv $(DESTDIR)/$(PREFIX)/bin'

? That would not have worked when I started this yesterday as (at that point) I didn't have a `PortGroup           makefile 1.0` line. However, I can imagine that maybe the existence of that line will mean that there are more env-vars in Make's envionment than I saw yesterday when PREFIX was not defined.

Reply:
@kesterlester personally I've simple use `destroot` block inside portfile to copy an artifact to a target.

Your way requires modify a Makefile from Portfile to adjust to real `${prefix}`.

Reply:
ok - thank you for the tip. I may try this later in the week.

Reply:
@kesterlester other way, probably cleaner, is using `DESTDIR` here.